### PR TITLE
Fix prometheus-critical-job-scrapping-failure alert

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

- fixes https://github.com/giantswarm/giantswarm/issues/25003 partially. We will have to investigate some of them
- 
### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Add Unit tests
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [ ] Alerting rules must have a comment documenting why it needs to exist.
- [ ] Consider creating a dashboard (if it does not exist already) to help oncallers monitor the status of the issue.

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 1]
```diff
@@ -65,19 +65,20 @@ spec:
...
         cancel_if_outside_working_hours: "true"
-
     - alert: PrometheusCriticalJobScrapingFailure
       annotations:
         description: {{`Prometheus {{$labels.installation}}/{{$labels.cluster_id}} has failed to scrape all targets in {{$labels.job}} job.`}}
         summary: Prometheus fails to scrape all targets in a job.
         opsrecipe: prometheus-job-scraping-failure/
-      expr: (count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics"} == 0) BY (app,job, installation, cluster_id) / count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics"}) BY (app, job, installation, cluster_id)) == 1
+      ## We ignore bastion hosts node exporters
+      expr: (count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics", job!~".*bastions.*"} == 0) BY (app,job, installation, cluster_id) / count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics", job!~".*bastions.*"}) BY (app, job, installation, cluster_id)) == 1
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:74]: Now we have UTs, it was easy to try this thing I've been thinking about for a while:
```suggestion
      expr: |-
        (
          count(
            up{
              app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics",
              job!~".*bastions.*"
            } == 0
          ) BY (app,job, installation, cluster_id)
          /
          count(
            up{
              app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics",
              job!~".*bastions.*"
            }
          ) BY (app, job, installation, cluster_id)
        ) == 1
```

WDYT? Isn't it much easier to read and debug?

**Review by hervenicol — APPROVED** [review]
LGTM.
Maybe it would have been nice to add some test cases for these problematic cases? May be a nice way to document all the strange conditions.

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml (hunk 1 above)
Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:74]: This looks amazing TBH

**Comment by QuentinBisson** [comment]
I'll try to add tests tomorrow then

**Pull request merged**